azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -41,12 +41,19 @@ def gen_transport_cert(self, prv_file, crt_file):
         """
         Create ssl certificate for https communication with endpoint server.
         """
-        cmd = ("{0} req -x509 -nodes -subj /CN=LinuxTransport -days 730 "
-               "-newkey rsa:2048 -keyout {1} "
-               "-out {2}").format(self.openssl_cmd, prv_file, crt_file)
-        rc = shellutil.run(cmd) # pylint: disable=C0103
-        if rc != 0:
-            logger.error("Failed to create {0} and {1} certificates".format(prv_file, crt_file))
+        cmd = [self.openssl_cmd, "req", "-x509", "-nodes", "-subj", "/CN=LinuxTransport", 
+            "-days", "730", "-newkey", "rsa:2048", "-keyout", prv_file, "-out", crt_file]
+        try:
+            shellutil.run_command(cmd)
+        except shellutil.CommandError as cmd_err:
+            msg = """Failed to create {0} and {1} certificates.
+            [stdout]
```

Comment:
Comment applies to everywhere you're using triple quotes: the indentation will be preserved in the string. Since the indentation varies depending on the source code, not the message, I would suggest removing all indentation from the string:
```
except shellutil.CommandError as cmd_err:
    msg = """Failed to decrypt {0}
[stdout]
{1}
[stderr]
{2}
""".format(p7m_file, stdout, stderr)
    logger.error(msg)
```